iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/CommandDetector.java

```diff
@@ -92,19 +104,34 @@ public List<Command> searchWithoutSplit(List<ArgumentResolution> resolvedArgumen
    * It will find only {@code echo} and {@code foo} and return as result.
    * <p>
    * This method split arguments at the beginning i.e.: {@code echo foo && echo bar} will be searched individually.
+   * It will store exported variable in exportedVariables and retain them in memory for the next commands., i.e.: {@code export MY_FLAG=true && command param},
+   * the "command param" will have knowledge about the variable 'MY_FLAG' and its value 'true'.
    */
   public List<Command> search(List<ArgumentResolution> resolvedArguments) {
+    var exportedVariables = new HashMap<String, String>();
     SeparatedList<List<ArgumentResolution>, String> splitCommands = splitCommands(resolvedArguments);
     List<Command> commands = new ArrayList<>();
 
     for (List<ArgumentResolution> resolved : splitCommands.elements()) {
-      commands.addAll(searchWithoutSplit(resolved));
+      if (!resolved.isEmpty() && isExportCommand(resolved.get(0))) {
+        exportedVariables.putAll(parseEnvShellVariable(resolved.subList(1, resolved.size())));
```

Comment:
That's a fair point; I didn't think about it.
I guess It can be added without breaking the current logic.

I'm not sure how useful it would be. Currently `CommandDetected` is never used for `export` command, and from my searches `export` commands can only be used to export variables and cannot be concatenated with other commands like we can have with "local" variables (`myvar=value command option`).

Reply:
I'm also not sure about the usefullness. Maybe just add a comment about it